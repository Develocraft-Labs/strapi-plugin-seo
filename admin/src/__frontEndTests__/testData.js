import trads from "../translations";

export const id = 1,
  firstname = "John",
  lastname = "Doe",
  email = "John.Doe@companyA.com",
  title = "title",
  created_by = {
    id,
    email,
    firstname,
    lastname,
  },
  created_at = "2021-10-12T11:49:01.042Z",
  updated_at = "2021-10-15T09:43:36.344Z",
  updated_by = { email, firstname, id, isActive: true, lastname },
  seo = {
    canonical: "https://www.companyA.com/articles",
    created_at: "2021-10-12T11:55:00.595Z",
    created_by: 1,
    id: 1,
    jsonLd: {},
    metaDescription: "Artists Desc",
    metaTitle: "Artists Title",
    ogImage: "Artist og image",
    ogTitle: "Artists Og Title",
    ogType: "Artists Og Type",
    ogUrl: "Artists og url",
    title: "Artists Seo",
    updated_at,
    updated_by: 1,
  },
  resourceData = {
    title,
    id,
    created_by,
  },
  collectionName = "Artists",
  pagination = { page: 1, total: 13, pageSize: 5, pageCount: 3 },
  uid = "application::artists.artists",
  fullResults = [
    { created_at, created_by, id, seo, title, updated_at, updated_by },
    { created_at, created_by, id: 2, seo, title, updated_at, updated_by },
    { created_at, created_by, id: 3, seo, title, updated_at, updated_by },
    { created_at, created_by, id: 4, seo, title, updated_at, updated_by },
    { created_at, created_by, id: 5, seo, title, updated_at, updated_by },
    { created_at, created_by, id: 6, seo, title, updated_at, updated_by },
    { created_at, created_by, id: 7, seo, title, updated_at, updated_by },
    { created_at, created_by, id: 8, seo, title, updated_at, updated_by },
    { created_at, created_by, id: 9, seo, title, updated_at, updated_by },
    { created_at, created_by, id: 10, seo, title, updated_at, updated_by },
    { created_at, created_by, id: 11, seo, title, updated_at, updated_by },
    { created_at, created_by, id: 12, seo, title, updated_at, updated_by },
    { created_at, created_by, id: 13, seo, title, updated_at, updated_by },
  ],
  seoId = 1,
  apiID = "artists",
  attributes = {
    seo: {
      dominant: false,
      nature: "oneWay",
      plugin: "seo",
      private: false,
      target: "plugins::seo.seo",
      unique: false,
    },
    title,
  },
  schema = {
    attributes,
    collectionName,
    connection: "default",
    description: "",
    draftAndPublish: true,
    kind: "collectionType",
    name: "Artists",
    pluginOptions: {},
    restrictRelationsTo: null,
    visible: true,
  },
  resourceSchema = { collectionName, schema },
  pluginUID = "plugins::seo.seo",
  projectCollectionTypes = [{ apiID, schema, uid }],
  selectedSeo = {
    canonical: "https://www.companyA.com/articles",
    created_at: "2021-10-12T11:55:00.595Z",
    created_by: created_by,
    id: 1,
    jsonLd: {},
    metaDescription: "Artists Desc",
    metaTitle: "Artists Title",
    ogImage: "Artist og image",
    ogTitle: "Artists Og Title",
    ogType: "Artists Og Type",
    ogUrl: "Artists og url",
    title: "Artists Seo",
    updated_at,
    updated_by: updated_by,
  },
  refreshOptions = () => {},
  metaTitle = "meta title",
  metaDescription = "meta description",
  canonical = "canonical",
  handleDeleteSeo = () => {},
  handleEditSeo = () => {},
  defaultLocale = "en",
  userEnabledLocales = [
    {
      code: "en",
      created_at: "2021-12-20T09:03:53.724Z",
      id: 1,
      isDefault: true,
      name: "English (en)",
      updated_at: "2021-12-20T09:03:53.724Z",
    },
  ],
  seos = [seo, fullResults];

export const resourceComponentTestData = {
  resourceData,
  resourceSchema,
};

export const seoFormComponentTestData = {
  pluginUID,
  seoId,
  projectCollectionTypes,
  resource: resourceComponentTestData,
};

export const contentTypesListComponentTestData = [
  {
    collectionName,
    fullResults,
    pagination,
    uid,
  },
];

export const translationPickerComponentTestData = {
  translations: trads,
  handleTranslation: () => {},
  value: "",
};

export const contentTypesItemComponentTestData = {
  uid,
  fullResults,
  pagination,
  collectionName,
};
export const previewComponentTestData = {
  selectedSeo,
  pluginUID,
  refreshOptions,
  projectCollectionTypes,
  handleDeleteSeo,
};

export const tableComponentTestData = {
  seos: { collectionName, fullResults, uid, pagination },
  handleDeleteSeo,
  handleEditSeo,
  userEnabledLocales,
  uid,
  defaultLocale,
};
