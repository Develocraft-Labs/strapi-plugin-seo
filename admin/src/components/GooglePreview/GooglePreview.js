import React, { useState } from "react";

import Desktop from "../Desktop/Desktop";
import Mobile from "../Mobile/Mobile";
import Radio from "../../ui/Inputs/Radio";
import shortenPreviewUrl from "../../utils/googlePreviewUrl";
import styled from "styled-components";
import { Row } from "../ui/common";

const DesktopWrap = styled.div`
  width: 290px;
  overflow-x: scroll;
`;

const PreviewContainer = styled.div`
  height: fit-content;
`;

const ModeList = styled(Row)`
  align-items: center;
  padding: 1rem 0rem 2rem 0rem;
`;

const PreviewMode = styled.h6`
  font-size: 1.5rem;
  padding: 2rem 0rem 0rem 0rem;
`;

const PreviewTitle = styled.h4`
  font-weight: 400;
`;

/**
 * Component that renders Google Preview.
 * @returns {JSX.Element} Either Mobile or Desktop versions.
 */
const GooglePreview = ({ title, metaDescription, ogUrl }) => {
  const [mobileValue, setMobileValue] = useState(true);
  const [desktopValue, setDesktopValue] = useState(false);

  const date = new Date();
  const month = date.toLocaleString("default", { month: "short" });
  const day = date.getUTCDate();
  const year = date.getUTCFullYear();

  const handleChange = (e) => {
    if (e.target.name === "mobile") {
      setMobileValue(true);
      setDesktopValue(false);
    } else {
      setMobileValue(false);
      setDesktopValue(true);
    }
  };

  return (
    <div>
      <PreviewTitle>Google Preview</PreviewTitle>
      <hr />
      <PreviewMode>Preview as:</PreviewMode>
      <ModeList onChange={handleChange}>
        <Radio
          name="mobile"
          label={"Mobile result"}
          id="mobile"
          checked={mobileValue}
        />
        <Radio
          name="desktop"
          label={"Desktop result"}
          id="desktop"
          checked={desktopValue}
          paddingLeft="1rem"
        />
      </ModeList>
      <PreviewContainer>
        {mobileValue ? (
          <Mobile
            title={title}
            metaDescription={metaDescription}
            ogUrl={shortenPreviewUrl(ogUrl, "mobile")}
            month={month}
            day={day}
            year={year}
          />
        ) : null}
        {desktopValue ? (
          <DesktopWrap>
            <Desktop
              title={title}
              metaDescription={metaDescription}
              ogUrl={shortenPreviewUrl(ogUrl, "desktop")}
              month={month}
              day={day}
              year={year}
            />
          </DesktopWrap>
        ) : null}
      </PreviewContainer>
    </div>
  );
};

export default GooglePreview;
