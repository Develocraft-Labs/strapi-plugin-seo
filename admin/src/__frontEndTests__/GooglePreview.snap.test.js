/**
 * @jest-environment jsdom
 */

import React from "react";
import renderer from "react-test-renderer";

import GooglePreview from "../components/GooglePreview/GooglePreview";
import { metaTitle, metaDescription, canonical } from "./testData";

describe("SNAP TESTING GOOGLE PREVIEW COMPONENT", () => {
  beforeAll(() => {
    jest.useFakeTimers("modern");
    jest.setSystemTime(new Date("2022-03-08"));
  });

  afterAll(() => {
    jest.useRealTimers();
  });

  it("Should Render Successfully", () => {
    const component = renderer.create(
      <GooglePreview
        metaTitle={metaTitle}
        metaDescription={metaDescription}
        canonical={canonical}
      />
    );
    const tree = component.toJSON();
    expect(component).toBeDefined();
    expect(tree).toMatchSnapshot();
  });
});
