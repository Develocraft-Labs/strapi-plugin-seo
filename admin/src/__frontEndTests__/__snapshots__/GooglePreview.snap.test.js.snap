// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SNAP TESTING GOOGLE PREVIEW COMPONENT Should Render Successfully 1`] = `
.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c15 {
  height: 2rem;
  width: 2rem;
}

.c16 {
  color: black;
  overflow-wrap: break-word;
}

.c14 {
  color: gray;
  font-size: 1.4rem;
}

.c13 {
  color: #7676fb;
  font-size: 1.6rem;
  overflow-wrap: break-word;
}

.c12 {
  padding: 0rem 0rem 0rem 1rem;
  font-size: 1.3rem;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0rem 0rem 1.3rem 0rem;
  overflow-wrap: break-word;
}

.c10 {
  box-shadow: 0 0px 10px #e3e9f3;
  padding: 1rem;
  border-radius: 1rem;
  width: 290px;
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
}

.c5 {
  width: 1.5rem;
  height: 1.5rem;
  cursor: pointer;
}

.c6 {
  margin: 0;
  padding: 0rem 0.5rem;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-left: 0rem;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-left: 1rem;
}

.c8 {
  height: 15rem;
}

.c3 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 1rem 0rem 2rem 0rem;
}

.c1 {
  font-size: 1.5rem;
  padding: 2rem 0rem 0rem 0rem;
}

.c0 {
  font-weight: 400;
}

<div>
  <h4
    className="c0"
  >
    Google Preview
  </h4>
  <hr />
  <h6
    className="c1"
  >
    Preview as:
  </h6>
  <div
    className="c2 c3"
    onChange={[Function]}
  >
    <div
      className="c4"
    >
      <input
        checked={true}
        className="c5"
        id="mobile"
        name="mobile"
        readOnly={true}
        type="radio"
        value="Mobile result"
      />
      <label
        className="c6"
        htmlFor="Mobile result"
      >
        Mobile result
      </label>
    </div>
    <div
      className="c7"
    >
      <input
        checked={false}
        className="c5"
        id="desktop"
        name="desktop"
        readOnly={true}
        type="radio"
        value="Desktop result"
      />
      <label
        className="c6"
        htmlFor="Desktop result"
      >
        Desktop result
      </label>
    </div>
  </div>
  <div
    className="c8"
  >
    <div
      className="c9 c10"
      data-testid="mobile-preview"
    >
      <small
        className="c11"
      >
        <i
          aria-hidden="true"
          className="fa fa-globe"
        />
         
        <span
          className="c12"
        />
      </small>
      <h3
        className="c13"
      />
      <div
        className="c14"
      >
        <span>
          Mar 8, 2022 
        </span>
        <svg
          className="c15"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            fill="gray"
            height="0.3rem"
            rx="2rem"
            ry="2rem"
            width="0.3rem"
            x="0.6rem"
            y="0.7rem"
          />
        </svg>
        <span
          className="c16"
        >
          meta description
        </span>
      </div>
    </div>
  </div>
</div>
`;
